src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private static final int INDEX_OPERATION = 0;
+    private static final int INDEX_FRUIT = 1;
+    private static final int INDEX_QUANTITY = 2;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        return inputReport.stream()
+                .map(string -> string.split(","))
+                .map(strings -> new FruitTransaction(
+                        strings[INDEX_OPERATION],
+                        strings[INDEX_FRUIT],
+                        strings[INDEX_QUANTITY]))
```

Comment:
The conversion from String to Integer for the quantity field is not being handled. The INDEX_QUANTITY is being used directly as a String, which will likely cause issues when attempting to work with the quantity as a numerical value. You should parse the quantity string to an integer and handle any potential NumberFormatException that might occur.

Reply:
++